projects/igniteui-angular/src/lib/grids/selection/selection.service.ts

```diff
@@ -609,9 +610,8 @@ export class IgxGridSelectionService {
         if (this.allRowsSelected !== undefined && !newSelection) {
             return this.allRowsSelected;
         }
-        const selectedData = new Set(newSelection ? newSelection : [...this.rowSelection]);
-        const allData = this.getRowIDs(this.allData);
-        const unSelectedRows = allData.filter(row => !selectedData.has(row));
+        const selectedData = new Set(newSelection ? newSelection.map(r => this.getRecordKey(r)) : [...this.rowSelection]);
+        const unSelectedRows = this.allData.filter(row => !selectedData.has(this.getRecordKey(row)) && !this.isRowDeleted(this.getRecordKey(row)));
```

Comment:
Looking at it again, is it needed to evaluate unselectedRows to include it in the calculation for all selected?

Once we have allData.length and selectedData.length === allData.length. I believe is enough to assume all rows are selected :) Let's try this